parlai/core/torch_agent.py

```diff
@@ -489,6 +493,33 @@ def add_cmdline_args(cls, argparser):
 
         cls.dictionary_class().add_cmdline_args(argparser)
 
+    @classmethod
+    def upgrade_saved_opt(cls, old_opt):
+        """
+        Upgrade an old opt file loaded from disk.
+
+        This aims to facilitate backwards compatibility of old model files, while
+        encouraging better defaults to be used in the future. Namely, it enables
+        for command line defaults to be set to one value, but old models on disk
+        to expect another default value.
+
+        Subclasses may *extend* this behavior, but should always call the
+        super-class.
+
+        :param old_opt:
+            The stored options, as loaded directly from disk.
+        :return:
+            The upgraded options. (May be inplace)
+        """
+
+        if 'add_delimiter_token' not in old_opt:
```

Comment:
this is a bit confusing since it only changes the flag 'add_delimiter_token' with no options to change anything else. is the intended behavior that models would override this?